### ISSUE TITLE
Style/StringLiterals is missing quoted quotes

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop#10120]
No offences for
```ruby
'hello \'world\'' # for EnforcedStyle: single_quotes
"hello \"world\"" # for EnforcedStyle: double_quotes
```
According to https://rubystyle.guide/#consistent-string-literals-single-quote, it's bad. It should be written as
```ruby
"hello 'world'"
'hello "world"'
```
instead.

## Describe the solution you'd like

Extend Style/StringLiterals to flag this.

**Comment by dvandersluis** [comment]
What would you expect to happen for 

```ruby
'hello "goodbye" \'world\''
```

**Comment by dvandersluis** [comment]
I have a change in progress but I'm wondering if this should be a new cop, because this is separate from ensuring quote consistency IMO. It might also be nice to have `'hello "goodbye" \'world\''` corrected to `%q{hello "goodbye" 'world'}`.

**Comment by dvandersluis** [comment]
@pirj clearly I lost track of this issue, and unfortunately I lost whatever code I was working on for this in 2021. However, I have written a new PR that adds a new cop to handle this. I went a slightly different route than the comments here, in that it enforces consistency of not escaping quotes in all cases. Please check it out!
